src/validations/html/runRules.js

```diff
@@ -0,0 +1,40 @@
+import {SAXParser} from 'parse5';
+import voidElements from 'void-elements';
+
+// Runs `rules` on `source` and promises an iterable of all errors.
+//
+// `rules` must be an array of objects that satisfy the interface:
+// * done() returns an iterable of errors
+// * optionally,
+//   openTag({row: number, column: number}, {name: string})
+//   to take in open tags that are neither void nor self-closing
+// * optionally,
+//   closeTag({row: number, column: number}, name: string)
+//   to take in close tags
+export default (rules, source) => {
+  const parser = new SAXParser({locationInfo: true});
+  return new Promise((resolve) => {
+    parser.on('startTag', (name, attrs, selfClosing, {line, col}) => {
```

Comment:
On `'startTag`, `SAXParser`'s [event listener](http://inikulin.github.io/parse5/classes/saxparser.html#on) takes in a positional `selfClosing` parameter.

Originally, I figured that it would be easier for the `MismatchedTag` rule to throw out whatever self-closing elements (valid or not) that occur. On the other hand, I don't think students would know about self-closing elements (or at least volunteers should probably avoid bringing them up) so for non-void elements, would it be better to assume that the element was intended to be a normal opening tag?